Compiler warnings on clang when using additional flags

### Version or commit hash

main

### RMW Implementation

FastRTPS

### Steps to reproduce issue

This is not a bug, nor a feature request. It is rather a small improvement to ensure safety and reduce warnings / errors for downstream users that compile with additional flags.

```
sudo apt install clang -y
# Build using clang
colcon build --cmake-args -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ - -DCMAKE_C_FLAGS='-Wformat -Werror=format-security' - -DCMAKE_CXX_FLAGS='-Wformat -Werror=format-security'

### Expected behavior

Compiles clean, no warnings.

### Actual behavior

Warnings are printed.

```bash
/craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/src/socket.cpp:88:24: warning: comparison of integers of different signs: 'int' and 'size_type' (aka 'unsigned long') [-Wsign-compare]
   88 |   if (last_newline_idx == string_buffer.size() - 1)
      |       ~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
In file included from /craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/src/trick_system.cpp:16:
In file included from /craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/include/trick_ros2_control/trick_system.hpp:26:
/opt/ros/jazzy/include/realtime_tools/realtime_tools/realtime_buffer.h:25:2: warning: "This header include is deprecated. Please update your code to use 'realtime_buffer.hpp' header." //NOLINT [-W#warnings]
   25 | #warning \
      |  ^
In file included from /craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/src/trick_system.cpp:16:
In file included from /craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/include/trick_ros2_control/trick_system.hpp:28:
/craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/include/trick_ros2_control/utils.hpp:27:76: warning: unused parameter 'delim' [-Wunused-parameter]
   27 | std::vector<std::string> trick_split_response(std::string& str, const char delim)
      |                                                                            ^
/craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/include/trick_ros2_control/utils.hpp:47:21: warning: comparison of integers of different signs: 'int' and 'size_type' (aka 'unsigned long') [-Wsign-compare]
   47 |   for (int i = 1; i < responseSplit.size(); i++)
      |                   ~ ^ ~~~~~~~~~~~~~~~~~~~~
/craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/src/trick_system.cpp:85:53: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
   85 |     RCLCPP_FATAL(rclcpp::get_logger("TrickSystem"), err.c_str());
      |                                                     ^~~~~~~~~~~
/opt/ros/jazzy/include/rclcpp/rclcpp/logging.hpp:1813:7: note: expanded from macro 'RCLCPP_FATAL'
 1813 |       __VA_ARGS__); \
      |       ^~~~~~~~~~~
/opt/ros/jazzy/include/rcutils/rcutils/logging_macros.h:1261:5: note: expanded from macro 'RCUTILS_LOG_FATAL_NAMED'
 1261 |     __VA_ARGS__)
      |     ^~~~~~~~~~~
/opt/ros/jazzy/include/rcutils/rcutils/logging_macros.h:81:73: note: expanded from macro 'RCUTILS_LOG_COND_NAMED'
   81 |       rcutils_log_internal(&__rcutils_logging_location, severity, name, __VA_ARGS__); \
      |                                                                         ^~~~~~~~~~~
/craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/src/trick_system.cpp:85:53: note: treat the string as an argument to avoid this
   85 |     RCLCPP_FATAL(rclcpp::get_logger("TrickSystem"), err.c_str());
      |                                                     ^
      |                                                     "%s", 
/opt/ros/jazzy/include/rclcpp/rclcpp/logging.hpp:1813:7: note: expanded from macro 'RCLCPP_FATAL'
 1813 |       __VA_ARGS__); \
      |       ^
/opt/ros/jazzy/include/rcutils/rcutils/logging_macros.h:1261:5: note: expanded from macro 'RCUTILS_LOG_FATAL_NAMED'
 1261 |     __VA_ARGS__)
      |     ^
/opt/ros/jazzy/include/rcutils/rcutils/logging_macros.h:81:73: note: expanded from macro 'RCUTILS_LOG_COND_NAMED'
   81 |       rcutils_log_internal(&__rcutils_logging_location, severity, name, __VA_ARGS__); \
      |                                                                         ^
/craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/src/trick_system.cpp:240:18: warning: format specifies type 'int' but the argument has type 'size_type' (aka 'unsigned long') [-Wformat]
  238 |                  "declared state variables. Expected: %d. "
      |                                                       ~~
      |                                                       %zu
  239 |                  "Got: %d",
  240 |                  trick_state_variables_.size(), raw_doubles_from_trick.size());
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/jazzy/include/rclcpp/rclcpp/logging.hpp:1813:7: note: expanded from macro 'RCLCPP_FATAL'
 1813 |       __VA_ARGS__); \
      |       ^~~~~~~~~~~
/opt/ros/jazzy/include/rcutils/rcutils/logging_macros.h:1261:5: note: expanded from macro 'RCUTILS_LOG_FATAL_NAMED'
 1261 |     __VA_ARGS__)
      |     ^~~~~~~~~~~
/opt/ros/jazzy/include/rcutils/rcutils/logging_macros.h:81:73: note: expanded from macro 'RCUTILS_LOG_COND_NAMED'
   81 |       rcutils_log_internal(&__rcutils_logging_location, severity, name, __VA_ARGS__); \
      |                                                                         ^~~~~~~~~~~
/craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/src/trick_system.cpp:240:49: warning: format specifies type 'int' but the argument has type 'size_type' (aka 'unsigned long') [-Wformat]
  239 |                  "Got: %d",
      |                        ~~
      |                        %zu
  240 |                  trick_state_variables_.size(), raw_doubles_from_trick.size());
      |                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/jazzy/include/rclcpp/rclcpp/logging.hpp:1813:7: note: expanded from macro 'RCLCPP_FATAL'
 1813 |       __VA_ARGS__); \
      |       ^~~~~~~~~~~
/opt/ros/jazzy/include/rcutils/rcutils/logging_macros.h:1261:5: note: expanded from macro 'RCUTILS_LOG_FATAL_NAMED'
 1261 |     __VA_ARGS__)
      |     ^~~~~~~~~~~
/opt/ros/jazzy/include/rcutils/rcutils/logging_macros.h:81:73: note: expanded from macro 'RCUTILS_LOG_COND_NAMED'
   81 |       rcutils_log_internal(&__rcutils_logging_location, severity, name, __VA_ARGS__); \
      |                                                                         ^~~~~~~~~~~
/craftsman_dev/src/robots/space_ros_demos/ros_trick/ros_src/trick_ros2_control/src/trick_system.cpp:243:21: warning: comparison of integers of different signs: 'int' and 'size_type' (aka 'unsigned long') [-Wsign-compare]
  243 |   for (int i = 0; i < raw_doubles_from_trick.size(); ++i)
      |                   ~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
6 warnings and 1 error generated.
```

### Backtrace or Console output

_No response_

--- a/ros_trick/ros_src/trick_ros2_control/include/trick_ros2_control/trick_system.hpp
+++ b/ros_trick/ros_src/trick_ros2_control/include/trick_ros2_control/trick_system.hpp
@@ -23,7 +23,7 @@
 #include "rclcpp/macros.hpp"
 #include "rclcpp/rclcpp.hpp"
 #include "rclcpp_lifecycle/state.hpp"
-#include "realtime_tools/realtime_buffer.h"
+#include "realtime_tools/realtime_buffer.hpp"
 #include "trick_ros2_control/socket.hpp"
 #include "trick_ros2_control/utils.hpp"
 #include "trick_ros2_control/visibility_control.h"

--- a/ros_trick/ros_src/trick_ros2_control/include/trick_ros2_control/utils.hpp
+++ b/ros_trick/ros_src/trick_ros2_control/include/trick_ros2_control/utils.hpp
@@ -24,7 +24,7 @@ double trick_string_convert(const std::string& str)
   return num;
 }
 
-std::vector<std::string> trick_split_response(std::string& str, const char delim)
+std::vector<std::string> trick_split_response(std::string& str, const char /*delim*/)
 {
   std::stringstream ss(str);
   std::string s;

--- a/ros_trick/ros_src/trick_ros2_control/include/trick_ros2_control/utils.hpp
+++ b/ros_trick/ros_src/trick_ros2_control/include/trick_ros2_control/utils.hpp
@@ -44,7 +44,7 @@ std::vector<double> trick_response_convert(std::string& response)
   {
     return result;
   }
-  for (int i = 1; i < responseSplit.size(); i++)
+  for (size_t i = 1; i < responseSplit.size(); i++)
   {
     result.push_back(trick_string_convert(responseSplit[i]));
   }

--- a/ros_trick/ros_src/trick_ros2_control/src/socket.cpp
+++ b/ros_trick/ros_src/trick_ros2_control/src/socket.cpp
@@ -81,8 +81,8 @@ std::string Socket::receive()
   }
 
   std::string string_buffer = std::string(carry_on_buffer_) + std::string(buffer);
-  int last_newline_idx = string_buffer.rfind(trick_delimiter_, string_buffer.size() - 1);
-  int second_last_newline_idx = string_buffer.rfind(trick_delimiter_, string_buffer.size() - 2);
+  auto last_newline_idx = string_buffer.rfind(trick_delimiter_, string_buffer.size() - 1);
+  auto second_last_newline_idx = string_buffer.rfind(trick_delimiter_, string_buffer.size() - 2);
 
   // it means that we can clear carry on buffer
   if (last_newline_idx == string_buffer.size() - 1)

--- a/ros_trick/ros_src/trick_ros2_control/src/trick_system.cpp
+++ b/ros_trick/ros_src/trick_ros2_control/src/trick_system.cpp
@@ -82,7 +82,7 @@ hardware_interface::CallbackReturn TrickSystem::on_configure(const rclcpp_lifecy
   {
     std::string err = "could not connect to Trick server on host: " + trick_hostname_ +
                       " and port: " + std::to_string(trick_server_port_);
-    RCLCPP_FATAL(rclcpp::get_logger("TrickSystem"), err.c_str());
+    RCLCPP_FATAL(rclcpp::get_logger("TrickSystem"), "%s", err.c_str());
     return hardware_interface::CallbackReturn::ERROR;
   }
 
@@ -235,8 +235,8 @@ hardware_interface::return_type TrickSystem::read(const rclcpp::Time& /*time*/, 
   {
     RCLCPP_FATAL(rclcpp::get_logger("TrickSystem"),
                  "size of data received from trick does not match number of "
-                 "declared state variables. Expected: %d. "
-                 "Got: %d",
+                 "declared state variables. Expected: %zu. "
+                 "Got: %zu",
                  trick_state_variables_.size(), raw_doubles_from_trick.size());
     return hardware_interface::return_type::OK;
   }

--- a/ros_trick/ros_src/trick_ros2_control/src/trick_system.cpp
+++ b/ros_trick/ros_src/trick_ros2_control/src/trick_system.cpp
@@ -240,7 +240,7 @@ hardware_interface::return_type TrickSystem::read(const rclcpp::Time& /*time*/, 
                  trick_state_variables_.size(), raw_doubles_from_trick.size());
     return hardware_interface::return_type::OK;
   }
-  for (int i = 0; i < raw_doubles_from_trick.size(); ++i)
+  for (size_t i = 0; i < raw_doubles_from_trick.size(); ++i)
   {
     double value_from_trick = raw_doubles_from_trick[i];
     double* const target_state_variable_ptr_ = trick_state_variables_[i].data_ptr;

